File: app/src/main/java/com/peterlaurence/trekadvisor/util/Tools.java
package com.peterlaurence.trekadvisor.util;

import java.io.PrintWriter;
import java.io.StringWriter;

/**
 * Utility class for general purposes.
 *
 * @author peterLaurence on 12/08/17.
 */
public class Tools {
    /**
     * Convert a {@link Throwable}'s stacktrace to a String.
     */
    public static String stackTraceToString(Throwable t) {
        StringWriter errors = new StringWriter();
        t.printStackTrace(new PrintWriter(errors));
        return errors.toString();
    }
}

File: app/src/main/java/com/peterlaurence/trekme/util/Tools.kt
package com.peterlaurence.trekme.util

import java.io.PrintWriter
import java.io.StringWriter

/**
 * Convert a [Throwable]'s stacktrace to a String.
 */
fun stackTraceToString(t: Throwable): String {
    val errors = StringWriter()
    t.printStackTrace(PrintWriter(errors))
    return errors.toString()
}

fun Throwable.stackTraceAsString(): String {
    val sw = StringWriter()
    printStackTrace(PrintWriter(sw))
    return sw.toString()
}
